lib/dynflow/director.rb

```diff
@@ -207,6 +212,25 @@ def track_execution_plan(execution_plan_id, finished)
       nil
     end
 
+    def update_execution_plan_state(manager)
+      execution_plan = manager.execution_plan
+      case execution_plan.state
+      when :running
+        if execution_plan.error?
+          execution_plan.execution_history.add('pause execution', @world.id)
+          execution_plan.update_state(:paused)
+        elsif manager.done?
+          execution_plan.execution_history.add('finish execution', @world.id)
+          execution_plan.update_state(:stopped)
+        end
+        # If the state is marked as running without errors but manager is not done,
+        # we let the invalidation procedure to handle re-execution on other executor
+      when :paused, :stopped
+        execution_plan.execution_history.add('finish execution', @world.id)
```

Comment:
Shouldn't it be `pause execution` when the execution plan gets paused?

Reply:
good catch. updated